c++/src/capnp/rpc-twoparty.h

```diff
@@ -83,7 +89,20 @@ class TwoPartyVatNetwork: public TwoPartyVatNetworkBase,
   class OutgoingMessageImpl;
   class IncomingMessageImpl;
 
-  kj::OneOf<kj::AsyncIoStream*, kj::AsyncCapabilityStream*> stream;
+  typedef kj::OneOf<
+      AsyncIoTransport,
+      AsyncCapabilityTransport,
+      MessageTransport*
+  > TransportUnion;
+
+  TransportUnion transport;
```

Comment:
I'd go with:

    MesasgeTransport& transport;
    kj::Own<MessageTransport> ownTransport;  // only if we created it

Saving an allocation here is not important as this is a long-lived object and many, many allocations will be performed while using it. (Though if you really wanted to save an allocation, you could use `kj::SpaceFor`... but I wouldn't.)

Or you could even reduce this to a single `kj::Own<MesasgeTransport> transport;` by using `NullDisposer` in the case that you don't actually own it... but that's a little hacky.